src/gui.rs

```diff
@@ -125,31 +126,27 @@ impl eframe::App for EsimMailerApp {
 
 impl EsimMailerApp {
     fn generate_preview(&mut self) {
-        let templates = crate::templates::load_templates();
-
-        if let Some(template) = templates.get("shared") {
-            let subject = crate::email::replace_placeholders(
-                template.subject,
-                &self.args.provider,
-                &self.args.name,
-                &self.args.data_amount,
-                &self.args.time_period,
-                &self.args.location,
-            );
-
-            let body = crate::email::replace_placeholders(
-                template.body,
-                &self.args.provider,
-                &self.args.name,
-                &self.args.data_amount,
-                &self.args.time_period,
-                &self.args.location,
-            );
-
-            self.email_preview = format!("Subject: {}\n\nBody:\n{}", subject, body);
-        } else {
-            self.email_preview = "Error: Shared template not found".to_string();
-        }
+        let template = get_template();
+
+        let subject = crate::email::replace_placeholders(
+            template.subject,
+            &self.args.provider,
+            &self.args.name,
+            &self.args.data_amount,
+            &self.args.time_period,
+            &self.args.location,
+        );
+
+        let body = crate::email::replace_placeholders(
+            template.body,
+            &self.args.provider,
+            &self.args.name,
+            &self.args.data_amount,
+            &self.args.time_period,
+            &self.args.location,
+        );
```

Comment:
Similar to my suggestion above, this could be a method on the template.
```suggestion
        let body = template.body(
            &self.args.provider,
            &self.args.name,
            &self.args.data_amount,
            &self.args.time_period,
            &self.args.location,
        );
```

Reply:
Updates have been pushed based on the conversation in the other thread.